### PR TITLE
Update File provider config API

### PR DESCRIPTION
**Pull request by vinser52**
### Description

Change the File Provider config as we agreed in #844.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly
- [x] New tests added, especially if they will fail without my changes

### CODE REVIEW

File: test/ipc_file_prov_consumer.c [hunk 1]
```diff
@@ -22,16 +22,35 @@ int main(int argc, char *argv[]) {
...
+            "[consumer] ERROR: creating File Memory Provider params failed\n");
+        return -1;
+    }
+
+    umf_result = umfFileMemoryProviderParamsSetVisibility(file_params,
+                                                          UMF_MEM_MAP_SHARED);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "[consumer] ERROR: setting File Memory Provider "
+                        "visibility failed\n");
+        goto destroy_provider_params;
```
Comment (ldorau) [inline test/ipc_file_prov_consumer.c:44]: Add `ret = -1;` here

File: test/ipc_file_prov_producer.c [hunk 2]
```diff
@@ -22,16 +22,35 @@ int main(int argc, char *argv[]) {
...
+            "[producer] ERROR: creating File Memory Provider params failed\n");
+        return -1;
+    }
+
+    umf_result = umfFileMemoryProviderParamsSetVisibility(file_params,
+                                                          UMF_MEM_MAP_SHARED);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "[producer] ERROR: setting File Memory Provider "
+                        "visibility failed\n");
+        goto destroy_provider_params;
```
Comment (ldorau) [inline test/ipc_file_prov_producer.c:44]: Add `ret = -1;` here

File: test/provider_file_memory.cpp [hunk 3]
```diff
@@ -163,23 +169,51 @@ TEST_F(test, test_if_mapped_with_MAP_SYNC) {
...
+    umf_result_t res = umfFileMemoryProviderParamsCreate(&file_params, path);
+    if (res != UMF_RESULT_SUCCESS) {
+        throw std::runtime_error(
+            "Failed to create File Memory Provider params");
+    }
+
+    res = umfFileMemoryProviderParamsSetVisibility(file_params,
+                                                   UMF_MEM_MAP_SHARED);
+    if (res != UMF_RESULT_SUCCESS) {
+        throw std::runtime_error("Failed to set visibility to shared for File "
```
Comment (ldorau) [inline test/provider_file_memory.cpp:202]: Add destroy params before

File: test/provider_file_memory_ipc.cpp [hunk 4]
```diff
@@ -17,39 +17,68 @@ using umf_test::test;
...
+    umf_result_t res = umfFileMemoryProviderParamsCreate(&file_params, path);
+    if (res != UMF_RESULT_SUCCESS) {
+        throw std::runtime_error(
+            "Failed to create File Memory Provider params");
+    }
+
+    res = umfFileMemoryProviderParamsSetVisibility(file_params,
+                                                   UMF_MEM_MAP_SHARED);
+    if (res != UMF_RESULT_SUCCESS) {
+        throw std::runtime_error("Failed to set visibility to shared for File "
```
Comment (ldorau) [inline test/provider_file_memory_ipc.cpp:35]: Add destroy params before

File: test/provider_file_memory_ipc.cpp [hunk 5]
```diff
@@ -17,39 +17,68 @@ using umf_test::test;
...
+    if (res != UMF_RESULT_SUCCESS) {
+        //test will be skipped.
+        return file_params_unique_handle_t(nullptr,
+                                           &umfFileMemoryProviderParamsDestroy);
+    }
+
+    res = umfFileMemoryProviderParamsSetVisibility(file_params,
+                                                   UMF_MEM_MAP_SHARED);
+    if (res != UMF_RESULT_SUCCESS) {
+        throw std::runtime_error("Failed to set visibility to shared for File "
```
Comment (ldorau) [inline test/provider_file_memory_ipc.cpp:58]: Add destroy params before

**Review by ldorau — CHANGES_REQUESTED** [review]

File: test/ipc_file_prov_producer.c (hunk 2 above)
Comment (vinser52) [reply test/ipc_file_prov_producer.c:44]: Done

File: test/provider_file_memory.cpp (hunk 3 above)
Comment (vinser52) [reply test/provider_file_memory.cpp:202]: Done

File: test/provider_file_memory_ipc.cpp (hunk 4 above)
Comment (vinser52) [reply test/provider_file_memory_ipc.cpp:35]: Done

File: test/provider_file_memory_ipc.cpp (hunk 5 above)
Comment (vinser52) [reply test/provider_file_memory_ipc.cpp:58]: Done

File: include/umf/providers/provider_file_memory.h [hunk 6]
```diff
@@ -18,15 +18,45 @@ extern "C" {
...
+/// @brief  Set the protection in the parameters struct.
+/// @param  hParams handle to the parameters of the File Memory Provider.
+/// @param  protection protection. Combination of \p umf_mem_protection_flags_t flags
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfFileMemoryProviderParamsSetProtection(
+    umf_file_memory_provider_params_handle_t hParams, unsigned protection);
+
+/// @brief  Set the visibility in the parameters struct.
+/// @param  hParams handle to the parameters of the File Memory Provider.
+/// @param  visibility visibility.
```
Comment (PatKamin) [inline include/umf/providers/provider_file_memory.h:55]: `memory visibility mode` would be a more useful description here.

**Review by ldorau — COMMENTED** [review]
The code coverage of the `src/provider/provider_file_memory.c` file on the main branch is **81%** while on this PR **78.8%**  - it could be improved, so that it does not decrease.

**Review by ldorau — APPROVED** [review]
The code coverage of the `src/provider/provider_file_memory.c` file on the main branch is **81%** while on this PR **78.8%**  - it could be improved, so that it does not decrease.

**Comment by vinser52** [comment]
> The code coverage of the `src/provider/provider_file_memory.c` file on the main branch is **81%** while on this PR **78.8%** - it could be improved, so that it does not decrease.

The main reason for code coverage decrease is because of these lines:
```c
#if defined(_WIN32) || defined(UMF_NO_HWLOC)

umf_memory_provider_ops_t *umfFileMemoryProviderOps(void) {
    // not supported
    return NULL;
}

umf_result_t umfFileMemoryProviderParamsCreate(
    umf_file_memory_provider_params_handle_t *hParams, const char *path) {
    (void)hParams;
    (void)path;
    return UMF_RESULT_ERROR_NOT_SUPPORTED;
}

umf_result_t umfFileMemoryProviderParamsDestroy(
    umf_file_memory_provider_params_handle_t hParams) {
    (void)hParams;
    return UMF_RESULT_ERROR_NOT_SUPPORTED;
}

umf_result_t umfFileMemoryProviderParamsSetPath(
    umf_file_memory_provider_params_handle_t hParams, const char *path) {
    (void)hParams;
    (void)path;
    return UMF_RESULT_ERROR_NOT_SUPPORTED;
}

umf_result_t umfFileMemoryProviderParamsSetProtection(
    umf_file_memory_provider_params_handle_t hParams, unsigned protection) {
    (void)hParams;
    (void)protection;
    return UMF_RESULT_ERROR_NOT_SUPPORTED;
}

umf_result_t umfFileMemoryProviderParamsSetVisibility(
    umf_file_memory_provider_params_handle_t hParams,
    umf_memory_visibility_t visibility) {
    (void)hParams;
    (void)visibility;
    return UMF_RESULT_ERROR_NOT_SUPPORTED;
}

#else // !defined(_WIN32) && !defined(UMF_NO_HWLOC)
```

We have no tests when provider is disabled at build time. And it is true for all providers.

**Comment by vinser52** [comment]
> > The code coverage of the `src/provider/provider_file_memory.c` file on the main branch is **81%** while on this PR **78.8%** - it could be improved, so that it does not decrease.
> 
> The main reason for code coverage decrease is because of these lines:
> 
> ```c
> #if defined(_WIN32) || defined(UMF_NO_HWLOC)
> 
> umf_memory_provider_ops_t *umfFileMemoryProviderOps(void) {
>     // not supported
>     return NULL;
> }
> 
> umf_result_t umfFileMemoryProviderParamsCreate(
>     umf_file_memory_provider_params_handle_t *hParams, const char *path) {
>     (void)hParams;
>     (void)path;
>     return UMF_RESULT_ERROR_NOT_SUPPORTED;
> }
> 
> umf_result_t umfFileMemoryProviderParamsDestroy(
>     umf_file_memory_provider_params_handle_t hParams) {
>     (void)hParams;
>     return UMF_RESULT_ERROR_NOT_SUPPORTED;
> }
> 
> umf_result_t umfFileMemoryProviderParamsSetPath(
>     umf_file_memory_provider_params_handle_t hParams, const char *path) {
>     (void)hParams;
>     (void)path;
>     return UMF_RESULT_ERROR_NOT_SUPPORTED;
> }
> 
> umf_result_t umfFileMemoryProviderParamsSetProtection(
>     umf_file_memory_provider_params_handle_t hParams, unsigned protection) {
>     (void)hParams;
>     (void)protection;
>     return UMF_RESULT_ERROR_NOT_SUPPORTED;
> }
> 
> umf_result_t umfFileMemoryProviderParamsSetVisibility(
>     umf_file_memory_provider_params_handle_t hParams,
>     umf_memory_visibility_t visibility) {
>     (void)hParams;
>     (void)visibility;
>     return UMF_RESULT_ERROR_NOT_SUPPORTED;
> }
> 
> #else // !defined(_WIN32) && !defined(UMF_NO_HWLOC)
> ```
> 
> We have no tests when provider is disabled at build time. And it is true for all providers.

Ok, I have added corresponding test.

**Comment by vinser52** [comment]
The new code coverage of the `src/provider/provider_file_memory.c` file is 81.7%.

File: include/umf/providers/provider_file_memory.h (hunk 6 above)
Comment (vinser52) [reply include/umf/providers/provider_file_memory.h]: Done

**Review by PatKamin — APPROVED** [review]

**Pull request merged**